lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -21,9 +29,9 @@ def lookup(path, vault_url = nil)
     use_ssl = uri.scheme == 'https'
     connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)
 
-    token = get_auth_token(connection)
+    token = get_auth_token(connection,vault_namespace)
 
-    secret_response = connection.get("/v1/#{path}", 'X-Vault-Token' => token)
+    secret_response = connection.get("/v1/#{path}", {'X-Vault-Token' => token, 'X-Vault-Namespace' => vault_namespace})
```

Comment:
Is is better to not set the header at all instead of an empty string?

Maybe something like...
```ruby
{'X-Vault-Token' => token, 'X-Vault-Namespace' => vault_namespace}.delete_if { |k,v| v.nil? }
```
and remove the
```ruby
if vault_namespace.nil?
  vault_namespace=''
end
```